libs/vault-api-client/src/Json.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\VaultApiClient;
+
+use JsonException;
+
+class Json
```

Comment:
Co pouzit `Symfony\Component\Serializer\Encoder` jako jinde?

Reply:
To by ale vyzadovalo sem pritahnout zavislost na `symfony/serializer`, coz mi prijde na druhou stranu jako overkill.

Ono cela ta `Json` class v tom Azure clientovi (odkud jsem kopiroval) puvodne vznikla hlavne proto, aby se navysilo score u Infection :smile: Aby se nemuselo vsude testovat porad dokola, jak kod failne, kdyz nedopadne dekodovani apod.

Reply:
Tak muzes pouzit `nette/utils`, kdyz ti symfony nevoni :trollface: 

Ja myslim ze to nevidim prvne nekde takovouhle classu. V nekterych komponentach/procesorech se pouziva to ze symfony. Mi prijde na pytel, ze se to pokazde dela jinak.

Reply:
> Mi prijde na pytel, ze se to pokazde dela jinak.

Dela se to na miru tak, jak je to tu potreba. Neumi to nic vic, nic min :slightly_smiling_face:

I kdyz tu pouziju `symfony/serializer`, skoncim u toho, ze z nej leze `mixed`, takze bud si udelam nad tim dalsi helper vrstvu, skrz kteoru to budu volat a ta mi to natypuje na `array`, nebo pri kazdym dekodovani (vim, jsou tu jen 2) budu checkovat `is_array`. Kdyz jedno z toho neudelam, vyfakuje me bud PHPstan, ze z toho muze vylez i jina vec nez `array`, nebo potom Infection ze nemam otesovany vsechny variace.

Vedl jsme podobnou debatu uz u toho Azure clienta s Odinem, tak se k tomu klidne muzem vratit pozdeji, az tu bude i on. Co me treba trha zily daleko vic je opacny pristup - na spouste mist napric code-base (hlavne v testech) pouzivame primo `json_decode` se vsema tema obskurnima parametrama a manualnim typovanim na `array` (protoze 99% situaci fakt ocekavame array) jen proto, abysme se nemuseli dohadovat o helperu :smile: